### PR TITLE
fix Issue 19812 - nested class reference causes dangling reference to…

### PR DESCRIPTION
**Pull request by WalterBright**
… out-of-scope struct

### CODE REVIEW

**Review by thewilsonator — APPROVED** [review]

**Comment by WalterBright** [comment]
Hmm, unexpected problems. I'll look at them tomorrow.

**Comment by WalterBright** [comment]
```
>>> TARGET FAILED: fail_compilation/test19812b.d
FAILED targets:
fail_compilation/test19812b.d
```
Ok, I am baffled. The test runs correctly on my local machine. The log says it failed, but the log does not show it being compiled. The only two references to test19812b.d are in the summary above.

I do not know how to proceed with this.

File: compiler/test/fail_compilation/test19812b.d [hunk 1]
```diff
@@ -0,0 +1,29 @@
+/* REQUIRED_ARGS=-preview=dip1000
```
Comment (dkorpel) [inline compiler/test/fail_compilation/test19812b.d:1]:
```suggestion
/* REQUIRED_ARGS: -preview=dip1000
```

> Test directive `REQUIRED_ARGS` must be followed by a `:`

Comment (WalterBright) [reply compiler/test/fail_compilation/test19812b.d]: Dang, right in front of my face! Thanks!

(But that doesn't explain why it doesn't appear in the log.)

Comment (dkorpel) [reply compiler/test/fail_compilation/test19812b.d]: It is printed to stdout, maybe it only shows stderr

Comment (dkorpel) [reply compiler/test/fail_compilation/test19812b.d]: I saw it printed when I ran it locally, did you not use the test runner? I don't understand how it ran correctly on your local machine, unless you manually compiled the test.

**Comment by WalterBright** [comment]
@atilaneves It's just your code on buildkite failing! Please post the offending sample if it isn't obvious what is going wrong.

**Comment by WalterBright** [comment]
@atilaneves -
```
tests/unit_threaded/ut/should.d(540,17): Error: reference to stack allocated value returned by `new C` assigned to non-scope parameter `value`
```

**Comment by atilaneves** [comment]
> @atilaneves -
> 
> ```
> tests/unit_threaded/ut/should.d(540,17): Error: reference to stack allocated value returned by `new C` assigned to non-scope parameter `value`
> ```

https://github.com/atilaneves/unit-threaded/pull/274

**Comment by WalterBright** [comment]
@atilaneves Ehhhxcellent!

**Comment by WalterBright** [comment]
@atilaneves still failing
```
tests/unit_threaded/ut/should.d(540,17): Error: reference to stack allocated value returned by `new C` assigned to non-scope parameter `value`
```

**Comment by Geod24** [comment]
It needs a new release, currently still picking up v2.1.0

**Comment by atilaneves** [comment]
Oops. Just tagged 2.1.1.

**Comment by WalterBright** [comment]
How about we delete std.logger.core until it starts working?
```
****** FAIL release32 std.logger.core
core.exception.AssertError@std/logger/core.d(2228): lineCall(2198) ll2Off(1) gll(1) ll(1) ll2(1) cond(0) condValue(0) memOrG(0) shouldLog(1) foo == 241 1 1 1 1 1
```

**Comment by WalterBright** [comment]
https://issues.dlang.org/show_bug.cgi?id=23286

**Review by RazvanN7 — APPROVED** [review]

File: druntime/src/object.d [hunk 2]
```diff
@@ -334,7 +334,7 @@ if ((is(LHS : const Object) || is(LHS : const shared Object)) &&
 /// This test passes `@safe` because it defines a new opEquals with `@safe`
 @safe unittest
 {
-    class F
+    static class F
```
Comment (ibuclaw) [inline druntime/src/object.d:337]: This is telling you there's a bug in this PR, why was it ignored?

**Pull request merged**